apis/v1alpha1/generator.yaml

```diff
@@ -17,12 +18,131 @@ ignore:
+  ListResourceRecordSets:
+    operation_type:
+    - List
+    resource_name:
+      RecordSet
   CreateHostedZone:
     output_wrapper_field_path: HostedZone
   GetHostedZone:
     output_wrapper_field_path: HostedZone
 prefix_config: {}
 resources:
+  RecordSet:
+    exceptions:
+      terminal_codes:
+      - InvalidChangeBatch
+      - InvalidInput
+      - NoSuchHostedZone
+      - NoSuchHealthCheck
+    fields:
+      AliasTarget:
+        from:
+          operation: ChangeResourceRecordSets
+          path: ChangeBatch.Changes.ResourceRecordSet.AliasTarget
+      # This field cannot be ignored, but values for this will be overriden and should not be used
+      ChangeBatch:
+        is_required: false
+      CidrRoutingConfig:
+        from:
+          operation: ChangeResourceRecordSets
+          path: ChangeBatch.Changes.ResourceRecordSet.CidrRoutingConfig
+      Failover:
+        from:
+          operation: ChangeResourceRecordSets
+          path: ChangeBatch.Changes.ResourceRecordSet.Failover
+      GeoLocation:
+        from:
+          operation: ChangeResourceRecordSets
+          path: ChangeBatch.Changes.ResourceRecordSet.GeoLocation
+      HealthCheckId:
+        from:
+          operation: ChangeResourceRecordSets
+          path: ChangeBatch.Changes.ResourceRecordSet.HealthCheckId
+      # Changing this value after a CR has been created could result in orphaned record sets
+      HostedZoneId:
+        references:
+          resource: HostedZone
+          path: Status.ID
+        is_required: true
+        is_immutable: true
+      ID:
+        is_primary_key: true
+        documentation: "ChangeID is used as a unique identifier"
+      MultiValueAnswer:
+        from:
+          operation: ChangeResourceRecordSets
+          path: ChangeBatch.Changes.ResourceRecordSet.MultiValueAnswer
+      # Changing this value after a CR has been created could result in orphaned record sets
+      Name:
+        from:
+          operation: ChangeResourceRecordSets
+          path: ChangeBatch.Changes.ResourceRecordSet.Name
+        is_required: true
+        is_immutable: true
+      # Changing this value after a CR has been created could result in orphaned record sets
+      RecordType:
+        from:
+          operation: ChangeResourceRecordSets
+          path: ChangeBatch.Changes.ResourceRecordSet.Type
+        is_required: true
+        is_immutable: true
+      Region:
+        from:
+          operation: ChangeResourceRecordSets
+          path: ChangeBatch.Changes.ResourceRecordSet.Region
+      ResourceRecords:
+        from:
+          operation: ChangeResourceRecordSets
+          path: ChangeBatch.Changes.ResourceRecordSet.ResourceRecords
+      # Changing this value after a CR has been created could result in orphaned record sets
+      SetIdentifier:
+        from:
+          operation: ChangeResourceRecordSets
+          path: ChangeBatch.Changes.ResourceRecordSet.SetIdentifier
+        is_immutable: true
```

Comment:
Can we make move it the status? or is this something the user can input. Could we consider randomizing it value?